tests/ReplSpec.hs

```diff
@@ -0,0 +1,49 @@
+-- |
+{-# LANGUAGE OverloadedStrings #-}
+
+module ReplSpec where
+
+import Test.Hspec
+
+import Data.ByteString (ByteString)
+import Control.Monad.IO.Class
+import qualified Data.ByteString as BS
+import Control.Concurrent (threadDelay)
+import System.Posix.Pty
+import Control.Monad (void)
+
+spec :: Spec
+spec = describe "ReplSpec" $ do
+  it "should not print literal via `Show` (regression #1101)" $ do
+    let src = "(print 1)"
+    out <- liftIO (runInteractive src)
+    out `shouldSatisfy` containsNoTermInfo
+
+  it "should not print `ErrInfo` via `Show` (regression #1164)" $ do
+    let src = "(module m g (defcap g () true) (defcap OFFERED (pid:string) true) \
+              \ (defpact sale () (step (= (create-capability-guard (OFFERED (pact-id)))\
+              \ (create-capability-guard (OFFERED pact-id)))))))"
+    out <- liftIO (runInteractive src)
+    out `shouldSatisfy`containsNoErrInfo
+
+
+runInteractive :: ByteString -> IO ByteString
```

Comment:
nit: Add some docstrings to explain the logic (e.g., mention how it is necessary to ignore the first `pact> `, and explain why we recursively call `readPty` rather than calling it once.